code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1187 @@
+		if(locked)
+			to_chat(user, "<span class='warning'>You need to open the shield before you can fix the mannequin.</span>")
+		else
+			if(health >= maxHealth)
+				to_chat(user, "<span class='warning'>Nothing to fix here!</span>")
+				return
+			var/obj/item/weapon/weldingtool/WT = W
+			if(WT.remove_fuel(5))
+				playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+				health = min(health + 20, maxHealth)
+				to_chat(user, "<span class='notice'>You fix some of the dents on \the [src]!</span>")
+			else
+				to_chat(user, "<span class='warning'>Need more welding fuel!</span>")
+				return
+
+	else if(istype(W, /obj/item/weapon/weldingtool) && user.a_intent == I_HELP)
+		if(locked)
+			to_chat(user, "<span class='warning'>You need to open the shield before you can fix the mannequin.</span>")
+		else
+			if(health >= maxHealth)
+				to_chat(user, "<span class='warning'>Nothing to fix here!</span>")
+				return
+			var/obj/item/weapon/weldingtool/WT = W
+			if(WT.remove_fuel(5))
+				playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+				health = min(health + 20, maxHealth)
+				to_chat(user, "<span class='notice'>You fix some of the dents on \the [src]!</span>")
+			else
+				to_chat(user, "<span class='warning'>Need more welding fuel!</span>")
+				return
+
+	else if(istype(W, /obj/item/device/silicate_sprayer))
+		if(!locked)
+			to_chat(user, "<span class='warning'>You need to lock the shield before you can fix it.</span>")
+		else
+			if(shield >= maxShield)
+				to_chat(user, "<span class='warning'>Nothing to fix here!</span>")
+				return
+			var/obj/item/device/silicate_sprayer/SS = W
+			if(SS.get_amount() >= 5)
+				SS.remove_silicate(5)
+				playsound(loc, 'sound/effects/refill.ogg', 50, 1)
+				shield = min(shield + 20, maxShield)
+				to_chat(user, "<span class='notice'>You fix some of the dents on \the [src]'s shield!</span>")
+			else
+				to_chat(user, "<span class='warning'>Need more silicate!</span>")
+				return
+
+	else if(user.a_intent == I_HURT)
+		user.delayNextAttack(8)
+		getDamage(W.force)
+		user.visible_message("<span class='danger'>[user.name] [W.attack_verb] \the [src]!</span>", "<span class='danger'>You [W.attack_verb] \the [src]!</span>")
+	else
+		return ..()
+
+/obj/structure/mannequin/cyber/update_icon()
+	overlays.len = 0
+	..()
+	if(destroyed)
+		overlays |= image(icon, "mannequin_cover_broken")
+	else if(locked)
+		overlays |= image(icon, "mannequin_cover")
+
+
+/obj/structure/mannequin/cyber/attack_hand(var/mob/living/user)
+	if(destroyed)
+		show_inv(user)
+	else if(locked)
+		if(user.a_intent == I_HURT)
+			user.delayNextAttack(8)
+			user.visible_message("<span class='danger'>[user.name] punches \the [src]!</span>", "<span class='danger'>You punch \the [src]!</span>", "You hear glass crack.")
+			getDamage(rand(1,7) * (user.get_strength() - 1))
+		else
+			to_chat(user,"<span class='notice'>You gently run your hands over \the [src] in appreciation of its contents.</span>")
```

Comment:
Celt, there's a difference between bad copypaste and code that exists twice. It's bad copypaste if they HAVE to be identical. In this case, it's just interaction. Could have been written from scratch.